Sources/Soto/Extensions/DynamoDB/DynamoDB+Codable.swift

```diff
@@ -371,10 +377,20 @@ extension DynamoDB {
             let expressionAttributeNames: [String: String]
             if let names = self.expressionAttributeNames, self.updateExpression != nil {
                 expressionAttributeNames = names
+            } else if let additionalAttributeNames {
+                let tmpAttributeNames: [String: String] = .init(item.keys.map { ("#\($0)", $0) }) { first, _ in return first }
+                expressionAttributeNames = tmpAttributeNames.merging(additionalAttributeNames, uniquingKeysWith: { _, new in new })
             } else {
                 expressionAttributeNames = .init(item.keys.map { ("#\($0)", $0) }) { first, _ in return first }
             }
-            let expressionAttributeValues: [String: AttributeValue] = .init(item.map { (":\($0.key)", $0.value) }) { first, _ in return first }
+
+            let expressionAttributeValues: [String: AttributeValue]
+            if let additionalAttributeValues {
```

Comment:
We still support swift 5.6, so this will need to be the old style `if let`

Reply:
https://github.com/soto-project/soto/pull/671/commits/94c7cf4290d42d6b9c34b42c097e5dee23f5c200